src/ZendViewRenderer.php

```diff
@@ -206,7 +213,13 @@ private function renderModel(ModelInterface $model, RendererInterface $renderer)
             }
 
             $child  = $this->mergeViewModel($child->getTemplate(), $child);
-            $result = $this->renderModel($child, $renderer);
+
+            if ($child !== $root) {
+                $viewModelHelper = $renderer->plugin('view_model');
```

Comment:
Shouldn't this be `$renderer->plugin(ViewModel::class);` where `ViewModel` is aliased to `\Zend\View\Helper\ViewModel`. At some point, all those old string aliases will go away won't they?

Reply:
That ist not correct. The ViewHelper is called 'view_model'. It is not using the container here, but the HelperPluginManager. You could call it with $renderer->view_model() also. But I prefer to call it this way. That's the way Zend_View does it internally.

Here is the code in PhpRenderer:
`            // Give view model awareness via ViewModel helper
            $helper = $this->plugin('view_model');
            $helper->setCurrent($model);
`

Or in Layout helper:
`    protected function getViewModelHelper()
    {
        if (null === $this->viewModelHelper) {
            $this->viewModelHelper = $this->getView()->plugin('view_model');
        }

        return $this->viewModelHelper;
    }
`

Reply:
@fduarte42 
> That ist not correct.

Please compare with:

https://github.com/zendframework/zend-view/blob/46128b668d0bf9f0abcfcf87be28e4e4927f0e02/src/HelperPluginManager.php#L188

And the aliases:

https://github.com/zendframework/zend-view/blob/46128b668d0bf9f0abcfcf87be28e4e4927f0e02/src/HelperPluginManager.php#L136-L139

> It is not using the container here, but the HelperPluginManager.

A plugin-manager is a service-manager / container:

https://docs.zendframework.com/zend-servicemanager/plugin-managers/

Reply:
Oh I see. I tried to use the class, but it does not work.
Perhaps we need to pump up the dependency of zend view to a higher version.
I have to investigaste it further.
And you are right. The HelperPluginManager is a Container. My bad.

Reply:
I use the class now. It is working now. I did import the worng class initially. Thank for correcting me. This is much better.